content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+That being said, it's important to note that Wolfi is rather new; it just recently crossed the mark of 1,000 packages in the Wolfi OS repository. That means some packages that you would find in a more established distro won't be available yet in Wolfi. In this article, we'll cover the whole process involved in building a new Wolfi package, or how a Wolfi package comes to be.
+
+Note: Many of the examples shown in this article are based on the [Wolfi PHP package](https://github.com/wolfi-dev/os/blob/main/php.yaml), which is a slightly complex build that generates several subpackages from a single melange YAML file. You can keep that link open in a separate tab to use as reference as you go through this guide.
+## How Does it Compile?
+The first step in building a new Wolfi package is finding official documentation with guidance on how to build the package from source. All Wolfi packages need to be built from source in order to assure provenance and authenticity of package contents.
+
+Because Wolfi uses [apk](https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper) and thus has some similar design principles as Alpine, it is a good idea to have a look at the [Alpine package index](https://pkgs.alpinelinux.org/packages) to find out how the package is built there. This can give you insights about configuration options, dependencies, and eventual subpackages that can be stripped from the main package. For example, when compiling PHP from source, you have the choice of compiling several extensions either as built-in or as shared libraries. Although compiling said extensions as built-in packages makes for a simpler build, it also increases the size of the original package and creates a wider surface for possible vulnerabilities.
+
+If you aren't very familiar with building packages from source using tools such as `cmake` and `autoconf`, it's a good idea to compile the package locally first - you don't need to run "make install"  at the end to get the package installed on your own system, but running the configure and make processes will give you a better understanding of the build requirements and configure options.
+## The melange YAML File
+The melange YAML file is where you'll define the details about the package and its build pipeline. If you are familiar with GitHub Actions, you'll find out that melange definitions are very similar to GitHub Actions workflows.
+### The `package` Section
+
+The melange YAML file starts with a `package` section, used to define metadata information and runtime dependencies. The following excerpt demonstrates how this section is declared in the Wolfi [PHP package](https://github.com/wolfi-dev/os/blob/34aa71ac4898c2b7c529548eafa51b0ea7a4dbd3/php.yaml#L1C3-L1C3) YAML:
+
+```yaml
+package:
+  name: php
+  version: 8.2.8
+  epoch: 0
+  description: the PHP programming language
+  copyright:
+	- license: PHP-3.01
```

Comment:
nit, but these seem to be tabs here, so for easier cut&paste, consider changing them to spaces. Most of the yaml sections seem to have it, but just jotting it here.

Reply:
yeah I think it got messed up when I copied the content from Google Docs. I will update the YAML sections to fix it

Reply:
I think it's all fixed now!